Create multitenancy.md (#1154)
cncf/glossary: content/en/multitenancy.md

---
title: Multitenancy
status: Feedback Appreciated
category: Property
---

## What it is

> Multitenancy (often written as multi-tenancy) refers to the capability of a single software installation to serve multiple tenants.  A tenant in this context is a user, application, or a group of users / applications that use the software to operate on their own set of data, and this data is only shared with other tenants as explicitly instructed by the owner – or, more often, not at all.  

Author: I guess the word "capability" here means that multitenant software may also serve just one tenant (let's say you installed it on-prem just for yourself...) but if it **_capable_** of serving multiple tenants, then it's already multitenant.

Tenants are normally not aware of each other, but they may be if they choose to be discoverable.   Say, GitHub is multitenant, but its whole point is enabing the tenants to cooperate.

> In other words, multitenant software provides each tenant with its segregated environment (work data, settings, list of credentials, etc.), simultaneously serving multiple tenants. From a tenant's perspective, each has its dedicated software installation, although, in reality, they are all sharing one. This is achieved by running the software on a server and allowing tenants to connect to it over the network via an interface and/or an [API](/application-programming-interface/) (also refer to [Client-Server Architecture](/client-server-architecture/)).

Reviewer: ```suggestion

```
Moved this into the how it works section

> 

Reviewer: ```suggestion

A tenant can be as small as one independent user with a single login ID — think personal productivity software — or as large as an entire corporation with thousands of login IDs, each with its own privileges yet interrelated in multiple ways. 
```
I would put this before the example

> In other words, multitenant software provides each tenant with their own segregated environment (work data, settings, list of credentials etc), serving multiple tenants at the same time and in a way that for each tenant it looks and feels as if they use their own dedicated installation of the software.  To achieve that, the software runs on a server and provides a way for the tenants to connect to it over the network, offering them a user interface and/or an [API](https://glossary.cncf.io/application-programming-interface) (see [Client-Server Architecture](https://glossary.cncf.io/client-server-architecture)).

Author: I agree

Multitenant software examples include Google Mail, Google Docs, Microsoft Office 365, Salesforce CRM, and Dropbox, among many more that categorize as either entirely or partially multitenant software. 

> A tenant can be as small as one independent user with a single login ID — think personal productivity software — or as large as an entire corporation with thousands of login IDs, each with its own privileges yet interrelated in multiple ways. 

Reviewer: ```suggestion

```
Moved this up

## Problem it addresses 

> Multitenancy is a key to operational efficiency of server-run software.  Providing each tenant with a dedicated installation is wasteful in terms of hardware utilization and maintenance efforts.  

Reviewer: ```suggestion
Without multitenancy, each tenant would need a dedicated software installation. This increases resource utilization and maintenance efforts, ultimately software costs.
```

## How it helps

> 

Reviewer: ```suggestion

Multitenant software provides each tenant a segregated environment (work data, settings, list of credentials, etc.), simultaneously serving multiple tenants. From a tenant's perspective, each has its dedicated software installation, although, in reality, they are all sharing one. This is achieved by running the software on a server and allowing tenants to connect to it over the network via an interface and/or an [API](/application-programming-interface/) (also refer to [Client-Server Architecture](/client-server-architecture/)).
```
This was in the first section and fits very well in here

> Multitenancy is a key to operational efficiency of server-run software.  Providing each tenant with a dedicated installation is wasteful in terms of hardware utilization and maintenance efforts.  

Author: I guess the first phrase can be moved to solution, yes.

> Multitenant software is designed in a way that tenants share the resources of one installation without affecting each other, or affecting only in predefined and controlled ways.  This results in huge savings for the software provider that can be passed on to the tenants, making services that would otherwise be very expensive much cheaper or even free (again, think web-based e-mail or document editors).

Reviewer: ```suggestion
With multitenant software, tenants share the resources of one installation without affecting each other or only in predefined and controlled ways. The resulting resource savings on the software provider's side can be passed on to the tenants, significantly reducing the software cost for users (again, think web-based e-mail or document editors).
```
Same content with some edits

## Related terms

> Multitenancy is not synonymous with [SaaS](https://glossary.cncf.io/software-as-a-service), although it is very common for SaaS to be multitenant and even to feature multitenancy as one of its core benefits.

Reviewer 2: ```suggestion
Multitenancy is not synonymous with [SaaS](/software-as-a-service/), although it is very common for SaaS to be multitenant and even to feature multitenancy as one of its core benefits.
```
